function-aws-api-proxy/src/main/java/io/micronaut/function/aws/proxy/MicronautLambdaContainerHandler.java

```diff
@@ -272,162 +306,118 @@ protected void handleRequest(
-                        }).block();
-                    } else {
-                        final Optional<UriRouteMatch<Object, Object>> finalRoute = lambdaContainerEnvironment.getRouter().route(
-                                containerRequest.getMethod(),
-                                containerRequest.getPath()
+                    @Override
+                    protected void doOnError(Throwable throwable) {
+                        final MutableHttpResponse<?> defaultErrorResponse = routeExecutor.createDefaultErrorResponse(request, throwable);
+                        encodeHttpResponse(
+                                request,
+                                response,
+                                defaultErrorResponse,
+                                defaultErrorResponse.body()
                         );
+                    }
 
-                        if (finalRoute.isPresent()) {
-                            final AnnotationMetadata annotationMetadata = finalRoute.get().getAnnotationMetadata();
-                            annotationMetadata.stringValue(Produces.class).map(MediaType::new)
-                                    .ifPresent(containerResponse::contentType);
+                    @Override
+                    protected void doOnComplete() {
+                    }
+                });
+    }
 
-                            final MediaType[] expectedContentType = Arrays.stream(annotationMetadata.stringValues(Consumes.class))
-                                    .map(MediaType::new)
-                                    .toArray(MediaType[]::new);
-                            final MediaType requestContentType = containerRequest.getContentType().orElse(null);
+    // Response is encoded/written in MicronautResponseWriter after handleRequest finishes,
+    // so we just need a complete MicronautAwsProxyResponse.
+    private void encodeHttpResponse(
+            MicronautAwsProxyRequest<?> request,
+            MicronautAwsProxyResponse<?> response,
+            HttpResponse<?> message,
+            Object body) {
+        toAwsProxyResponse(response, message);
+    }
 
-                            if (expectedContentType.length > 0 && Arrays.stream(expectedContentType).noneMatch(ct -> ct.equals(requestContentType)) && Arrays.stream(expectedContentType).noneMatch(ct -> ct.equals(MediaType.ALL_TYPE))) {
-                                containerResponse.status(HttpStatus.UNSUPPORTED_MEDIA_TYPE);
-                                containerResponse.close();
-                                return;
-                            }
-                        }
+    private MicronautAwsProxyResponse<?> toAwsProxyResponse(
+            MicronautAwsProxyResponse<?> response,
+            HttpResponse<?> message) {
 
-                        final Optional<URL> staticMatch = resourceResolver.resolve(containerRequest.getPath());
-                        if (staticMatch.isPresent()) {
```

Comment:
I think this change has introduced a regression which stops the swagger integration from working since the documentation states it uses static resources.

i.e.
```
micronaut:
    router:
        static-resources:
            swagger:
                paths: classpath:META-INF/swagger
                mapping: /swagger/**
```

If there is an alternative for how this is meant to work, then the documentation needs to be updated.

Reply:
can you create an issue which steps to reproduce?

Reply:
Hi, I'm running into this issue with generic static resources, not particularly swagger. I'm using micronaut-parent 3.7.4 and when I run the app locally, static resources work fine. When I deploy the app to AWS Lambda, I get a 404 for the same request(s). Why was this code removed?

Reply:
Is there any ETA on this issue?